src/ol/layer/Graticule.js

```diff
@@ -479,11 +479,25 @@ class Graticule extends VectorLayer {
    * @return {Array<import("../extent.js").Extent>} Extents.
    */
   strategyFunction(extent, resolution) {
-    if (this.loadedExtent_ && !equals(this.loadedExtent_, extent)) {
+    // extents may be passed in different worlds, to avoid endless loop we use only one
+    const realExtent = extent.slice();
+    if (this.projection_) {
+      const center = getCenter(extent);
+      const projectionExtent = this.projection_.getExtent();
+      const worldWidth = getWidth(projectionExtent);
+      if (this.getSource().getWrapX() && this.projection_.canWrapX() && !containsExtent(projectionExtent, extent)) {
+        const worldsAway = Math.floor((center[0] - projectionExtent[0]) / worldWidth);
+        realExtent[0] -= (worldsAway * worldWidth);
+        realExtent[2] -= (worldsAway * worldWidth);
+      }
+      realExtent[0] = Math.round(realExtent[0] * 1e6) / 1e6;
```

Comment:
Better use the resolution to round to pixels, instead of hard-coding `1e6`.

Reply:
I don't think it is related to resolution. There is a javascript rounding problem when adding or subtracting multiples of the width which could happen at any resolution breaking the equals check and therefore not fixing the endless loop
![image](https://user-images.githubusercontent.com/49240900/77923341-3badd700-729a-11ea-88bf-b5a4b16bebdf.png)